src/SanityCheck/CoherenceVerifier.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Versioning;
+using NuGet;
+
+namespace SanityCheck
+{
+    public class CoherenceVerifier
+    {
+        public static bool VerifyAll(ProcessResult result)
+        {
+            foreach (var productPackageInfo in result.ProductPackages.Values)
+            {
+                Visit(productPackageInfo, result);
+            }
+
+            var success = true;
+            foreach (var packageInfo in result.ProductPackages.Values)
+            {
+                if (!packageInfo.Success)
+                {
+                    // Temporary workaround for FileSystemGlobbing used in Runtime.
+                    if (packageInfo.Package.Id.Equals("Microsoft.Extensions.Runtime", StringComparison.OrdinalIgnoreCase) &&
+                        packageInfo.DependencyMismatches.All(d => d.Dependency.Id.Equals("Microsoft.Extensions.FileSystemGlobbing", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        continue;
+                    }
+
+                    // Temporary workaround for xunit.runner.aspnet used in Microsoft.AspNet.Testing.
+                    if (packageInfo.Package.Id.Equals("Microsoft.AspNet.Testing", StringComparison.OrdinalIgnoreCase) &&
+                        packageInfo.DependencyMismatches.All(d => d.Dependency.Id.Equals("xunit.runner.aspnet", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        continue;
+                    }
+
+                    if (packageInfo.InvalidCoreCLRPackageReferences.Count > 0)
+                    {
+                        Log.WriteError("{0} has invalid package references:", packageInfo.Package.GetFullName());
+
+                        foreach (var invalidReference in packageInfo.InvalidCoreCLRPackageReferences)
+                        {
+                            Log.WriteError("Reference {0}({1}) must be changed to be a frameworkAssembly.",
+                                invalidReference.Dependency,
+                                invalidReference.TargetFramework);
+                        }
+                    }
+
+                    if (packageInfo.DependencyMismatches.Count > 0)
+                    {
+                        Log.WriteError("{0} has mismatched dependencies:", packageInfo.Package.GetFullName());
+
+                        foreach (var mismatch in packageInfo.DependencyMismatches)
+                        {
+                            Log.WriteError("    Expected {0}({1}) but got {2}",
+                                mismatch.Dependency,
+                                (mismatch.TargetFramework == VersionUtility.UnsupportedFrameworkName ?
+                                "DNXCORE50" :
```

Comment:
Also I think that's a vestige from using a NuGet.core that did not support DNXCore50. I'll leave it around for now